### PR TITLE
Update examples.

### PR DESCRIPTION
**Pull request by manthey**
Tweak a few examples to fix some inconsistencies and errors.  The goal is for every example to start in an interesting spot, and, moreover, for the thumbnail of the example to be representative of the example.

Added an optional `thumbquery` parameter to example.json files.  If present, this should be added to the base url when generating the thumbnail for the example.

Adjusted the zoom and center of several examples to start in more interesting locations.

Fixed some style issues.

If a heatmap was panned before it was ready, it could throw an error.

If a reprojection's viewport corners were ill-defined in the projection, the map would fail to draw.

For the WMS and SLD examples, switch the URL used for the geoserver to https.  This works -- the http server no longer does (this resolves issue #690).

For the SLD example, add controls to allow setting the base url, layer name, and projection.  Clean up some of the pug code.  Update the thumbnail to match what the default settings show.

### CODE REVIEW

File: examples/sld/main.js [hunk 1]
```diff
@@ -12,40 +12,36 @@ var layer = {
   projection: 'EPSG:3785'
 };
 
+var baseUrl = 'https://demo.boundlessgeo.com/geoserver/ows';
```
Comment (aashish24) [inline examples/sld/main.js]: +1

File: src/tileLayer.js [hunk 2]
```diff
@@ -1416,16 +1416,23 @@ module.exports = (function () {
...
           mapzoom = map.zoom(),
           roundzoom = this._options.tileRounding(mapzoom),
-          unit = map.unitsPerPixel(zoom === undefined ? roundzoom : zoom);
+          unit = map.unitsPerPixel(zoom === undefined ? roundzoom : zoom),
+          gcsPt;
       if (pt === undefined) {
         var size = map.size();
         pt = {x: size.width / 2, y: size.height / 2};
       }
+      /* displayToGcs can fail under certain projections.  If this happens,
```
Comment (aashish24) [inline src/tileLayer.js]: how can we replicate this behavior in our current version?

Comment (aashish24) [reply src/tileLayer.js]: just curious but agree on the statement

Comment (manthey) [reply src/tileLayer.js]: In the reprojection example, choose a projection that curves at the edges, (ESRI:53009 or any Mo
llweide will work).  Zoom out.  If you zoomed out far enough, this code path is triggered.

Comment (aashish24) [reply src/tileLayer.js]: Got it, thanks.

**Pull request merged**